data/core/contextmenu.lua

```diff
@@ -88,6 +88,7 @@ function ContextMenu:show(x, y)
     end
     if y + h >= core.root_view.size.y then
       y = y - h
+      if y < 0 then y = 0 end
```

Comment:
What about `common.clamp` for this?

Reply:
was thinking how to substitute the whole 4 lines if with a `common.clamp` but couldn't come with any solution (mind went white xD).

Reply:
???
```lua
y = common.clamp(y, 0, core.root_view.size.y - h)
```

Reply:
This might just work.

Also isn't the original wrong?
Shouldn't it have been `y = core.root_view.size.y - h` instead of `y = y - h`?
Same for `x`.

Reply:
ok, did for both x and y, the only drawback is we are now performing 6 function calls, but this part isn't performance critical :)